src/eval/operation.rs

```diff
@@ -2513,6 +2531,71 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                     ))
                 }
             }
+            BinaryOp::DictionaryAssume() => {
+                let (
+                    Closure {
+                        body: contract_term,
+                        env: contract_env,
+                    },
+                    _,
+                ) = self.stack.pop_arg(&self.cache).ok_or_else(|| {
+                    EvalError::NotEnoughArgs(3, String::from("dictionary_assume"), pos_op)
+                })?;
+
+                let lbl = match_sharedterm! {t1, with {
+                        Term::Lbl(lbl) => lbl
+                    } else return Err(EvalError::TypeError(
+                        String::from("Lbl"),
+                        String::from("dictionary_assume, 2nd argument"),
+                        fst_pos,
+                        RichTerm {
+                            term: t1,
+                            pos: pos1,
+                        }
+                    ))
+                };
+
+                match_sharedterm! {t2,
+                    with {
+                        Term::Record(record_data) => {
+                            let contract = contract_term.closurize(&mut self.cache, &mut env2, contract_env);
+                            let record_contract = Term::Record(RecordData{
+                                fields: record_data.fields.iter().map(|(key, value)| (*key, Field {
+                                    value: None,
+                                    pending_contracts: vec![PendingContract::new(contract.clone(), lbl.clone())],
+                                    metadata: FieldMetadata {
+                                        opt: value.metadata.opt,
+                                        ..Default::default()
+                                    },
+                                })).collect(),
+                                attrs: RecordAttrs{
+                                    open: true
+                                },
+                                ..Default::default()
+                            });
+
+                            let merge_term = Closure {
+                                body: mk_opn!(
+                                    NAryOp::MergeContract(),
+                                    Term::Lbl(lbl),
+                                    Term::Record(record_data),
+                                    record_contract
+                                ),
+                                env: env2,
+                            };
+                            Ok(merge_term)
+                        }
+                    } else Err(EvalError::TypeError(
```

Comment:
I didn't even know you could omit the `{ ... }` around the else block :exploding_head:

Reply:
🤯 I don't think I did that conciously

Reply:
Oh, I think that's a quirk in the `match_sharedterm!` macro. It expects the else clause to be an `expr`. So it can be either a single expression, like it is here, or a compound block with `{ ... }`. This doesn't work with if-else in rust.

Reply:
Yep, I saw that this occurs in a `matched_sharedterm`, but I still thought that one needed braces. It just turned out this was an habit.